### PR TITLE
chore: refactor SparklineCell for react 17 and react 18 upgrade

### PR DESCRIPTION
**Pull request by lilykuang**
### SUMMARY

@data-ui/sparkline doesn't support react 17 or react 18 
- remove @data-ui/sparkline
- migrate from @data-ui/sparkline to visx/xychart for SparklineCell

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

After
SparklineCell

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [ ] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

File: superset-frontend/src/visualizations/TimeTable/SparklineCell.tsx [hunk 1]
```diff
@@ -18,39 +18,36 @@
...
 interface Props {
-  ariaLabel: string;
+  dataKey: string;
   className?: string;
+  data: Array<number>;
+  entries: Array<any>;
   height: number;
-  numberFormat: string;
-  renderTooltip: ({ index }: { index: number }) => void;
+  numberFormat?: string;
```
Comment (Antonio-RiveroMartnez) [inline superset-frontend/src/visualizations/TimeTable/SparklineCell.tsx:45]: This wasn't optional before, any reason why we want it to be optional now?

File: superset-frontend/src/visualizations/TimeTable/SparklineCell.tsx [hunk 2]
```diff
@@ -18,39 +18,36 @@
...
+  dateFormat?: string;
```
Comment (Antonio-RiveroMartnez) [inline superset-frontend/src/visualizations/TimeTable/SparklineCell.tsx:46]: This is new, but same as with `numberFormat`, do we want this to be optional?

File: superset-frontend/src/visualizations/TimeTable/SparklineCell.tsx [hunk 3]
```diff
@@ -18,39 +18,36 @@
...
-  WithTooltip,
-} from '@data-ui/sparkline';
-import { getTextDimension, formatNumber } from '@superset-ui/core';
+  Tooltip,
+  XYChart,
+  buildChartTheme,
+} from '@visx/xychart';
 
 interface Props {
-  ariaLabel: string;
```
Comment (Antonio-RiveroMartnez) [inline superset-frontend/src/visualizations/TimeTable/SparklineCell.tsx:32]: Could we instead keep using it and pass it as the `accessibilityLabel` of the `XYChart`? https://airbnb.io/visx/docs/xychart#XYChart_accessibilityLabel

**Review by Antonio-RiveroMartnez — APPROVED** [review]
LGTM!

File: superset-frontend/package.json [hunk 4]
```diff
@@ -114,7 +113,12 @@
     "@superset-ui/plugin-chart-word-cloud": "file:./plugins/plugin-chart-word-cloud",
     "@superset-ui/preset-chart-xy": "file:./plugins/preset-chart-xy",
     "@superset-ui/switchboard": "file:./packages/superset-ui-switchboard",
+    "@visx/axis": "^3.0.1",
+    "@visx/grid": "^3.0.1",
```
Comment (eschutho) [inline superset-frontend/package.json:117]: any reason for/against using the same version across all @vizx packages?

Comment (lilykuang) [reply superset-frontend/package.json:117]: oh good catch. will update all visx to 3.0.1

Comment (lilykuang) [reply superset-frontend/package.json:117]: tried to install @visx/xychart 3.0.1 but there was a react-spring error

**Review by eschutho — APPROVED** [review]

**Pull request merged**